mesa_crm_lead_projects/models/crm/lead.py

```diff
@@ -0,0 +1,37 @@
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    project_ids = fields.One2many("project.project", "opportunity_id")
+    project_id = fields.Many2one(
+        "project.project", compute="_compute_project_id", string="Project"
+    )
+    project_count = fields.Integer(
+        compute="_compute_project_count", string="Project Card"
+    )
+
+    @api.multi
+    @api.depends("project_ids")
+    def _compute_project_count(self):
+        """
+        Count the number of Projects associated with the Lead.
+
+        :return {NoneType}:
+        """
+        for crm_lead in self:
+            crm_lead.project_count = len(crm_lead.project_ids)
+
+    @api.multi
+    @api.depends("project_ids")
+    def _compute_project_id(self):
+        """
+        Compute the Project Field to be the first Project in the list of
+        Projects associated with this Lead/Opportunity.
+
+        :return {NoneType}
+        """
+        for crm_lead in self:
+            if crm_lead.project_ids:
+                crm_lead.project_id = crm_lead.project_ids[0].id
```

Comment:
Would something like this work?  The `filtered()` in the `for` cuts out the leads that don't have any projects.  The switch from `[0]` to `[:1]` just makes sure that there isn't an error if there isn't a record.

```suggestion
        for crm_lead in self.filtered('project_ids'):
            crm_lead.project_id = crm_lead.project_ids[:1].id
```

Reply:
This works exactly the same - is there any performance benefit to this?

Reply:
Potentially less iterations in the `for` loop.  If you do it the other way, then the `for` loop iterates over all of the leads whether they have projects or not.